### PR TITLE
Update cni test helper to multi-arch image

### PR DESCRIPTION
**Pull request by vikasmb**
- Also changed the weekly test workflow to execute all steps even if an earlier step fails

**What type of PR is this?**
feature

**Which issue does this PR fix**:
Builds multi-arch image for CNI test helper.

**What does this PR do / Why do we need it**:
Builds multi-arch image for CNI test helper.
Also, executes all steps in weekly test workflow despite any failures in previous steps.

**If an issue # is not available please add repro steps and logs from IPAMD/CNI showing the issue**:
#1842 

**Testing done on this change**:

```
ginkgo -v --focus="CANARY" --failOnPending --  --cluster-kubeconfig=$KUBECONFIG  --cluster-name=$CLUSTER_NAME  --aws-region=$AWS_REGION  --aws-vpc-id=$VPC_ID  --ng-name-label-key=$NG_NAME_LABEL_KEY  --ng-name-label-val=$NG_NAME_LABEL_VAL
```
was successful with the new test helper image.

**Automation added to e2e**:

**Will this PR introduce any new dependencies?**:

No
**Will this break upgrades or downgrades. Has updating a running cluster been tested?**:
No

**Does this change require updates to the CNI daemonset config files to work?**:

No
**Does this PR introduce any user-facing change?**:

No
```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jayanthvn — APPROVED** [review]
lgtm

File: test/agent/go.mod [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-go 1.14
+go 1.17
 
 require (
 	github.com/coreos/go-iptables v0.6.0
 	github.com/vishvananda/netlink v1.1.0
 	golang.org/x/sys v0.0.0-20210426230700-d19ff857e887
 )
+
+require github.com/vishvananda/netns v0.0.0-20191106174202-0a2b9b5464df // indirect
```
Comment (achevuru) [inline test/agent/go.mod:11]: Latest not compatible with go 1.17?

Comment (vikasmb) [reply test/agent/go.mod:11]: This line got inserted upon `go mod tidy`, it seems like both https://github.com/vishvananda/netlink and github.com/vishvananda/netns specify go 1.12 version.

**Review by orsenthil — APPROVED** [review]
LGTM.

File: .github/workflows/weekly-cron-test.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ name: Weekly CNI tests
 
 on:
   schedule:
-    - cron: "0 16 * * 2"  # every Tuesday
+    - cron: "0 22 * * 3"  # every Wednesday
```
Comment (orsenthil) [inline .github/workflows/weekly-cron-test.yml:5]: Curious, why this change?

Comment (vikasmb) [reply .github/workflows/weekly-cron-test.yml:5]: We would want to execute all steps in weekly perf test despite failures of any one step in the workflow. Followed suggestion in https://github.community/t/continue-on-error-allow-failure-ui-indication/16773/2 to achieve this functionality. 

Change in cron schedule here is to test the effect of this workflow change today.
